pkg/tests/schema_pulumi_test.go

```diff
@@ -1508,3 +1509,263 @@ runtime: yaml
 		})
 	}
 }
+
+func TestConfigureGetRaw(t *testing.T) {
+	getOkExists := func(d *schema.ResourceData, key string) (interface{}, bool) {
+		v := d.GetRawConfig().GetAttr(key)
+		if v.IsNull() {
+			return nil, false
+		}
+		return d.Get(key), true
+	}
+	resMap := map[string]*schema.Resource{
+		"prov_test": {
+			Schema: map[string]*schema.Schema{
+				"test": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+		},
+	}
+
+	runConfigureTest := func(t *testing.T, configPresent bool) {
+		tfp := &schema.Provider{
+			ResourcesMap: resMap,
+			Schema: map[string]*schema.Schema{
+				"config": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+			ConfigureContextFunc: func(ctx context.Context, rd *schema.ResourceData) (interface{}, diag.Diagnostics) {
+				_, ok := getOkExists(rd, "config")
+				require.Equal(t, configPresent, ok, "Unexpected config value")
+				return nil, nil
+			},
+		}
+		bridgedProvider := pulcheck.BridgedProvider(t, "prov", tfp)
+		configVal := "val"
+		if !configPresent {
+			configVal = "null"
+		}
+		program := fmt.Sprintf(`
+name: test
+runtime: yaml
+resources:
+  prov:
+    type: pulumi:providers:prov
+	defaultProvider: true
+	properties:
+	  config: %s
+  mainRes:
+    type: prov:index:Test
+	properties:
+	  test: "hello"
+outputs:
+  testOut: ${mainRes.test}
+`, configVal)
+		pt := pulcheck.PulCheck(t, bridgedProvider, program)
+		res := pt.Up()
+		require.Equal(t, "hello", res.Outputs["testOut"].Value)
```

Comment:
This is a weak assert for this test, how do we know that the provider received the right data in `d.GetRawConfig()`?

Reply:
This is a smoke test really - this used to panic before the fix. The other test is the one which is meant to test the equivalence of the config to TF.

I'll rename the test to make that clearer

Reply:
Yeah, appreciated.